components/connectivity-adapter/internal/appregistry/service/converter.go

```diff
@@ -174,39 +168,145 @@ func (c *converter) DetailsToGraphQLInput(deprecated model.ServiceDetails) (grap
 			}
 		}
-		out.APIDefinitions = []*graphql.APIDefinitionInput{outApi}
-
 	}
 
 	if deprecated.Events != nil && deprecated.Events.Spec != nil {
-		out.EventDefinitions = []*graphql.EventDefinitionInput{
-			{
+		out.Event =
+			&graphql.EventDefinitionInput{
 				Spec: &graphql.EventSpecInput{
 					Data: ptrClob(graphql.CLOB(deprecated.Events.Spec)),
 				},
-			},
-		}
-	}
+			}
 
-	if deprecated.Documentation != nil {
-		// TODO later
 	}
 
-	if deprecated.Labels != nil && *deprecated.Labels != nil {
-		for k, v := range *deprecated.Labels {
-			outLabels[k] = v
+	return out, nil
+}
+
+func (c *converter) GraphQLToServiceDetails(in model.GraphQLServiceDetails) (model.ServiceDetails, error) {
+	outDeprecated := model.ServiceDetails{}
+	if in.API != nil {
+		outDeprecated.Api = &model.API{
+			TargetUrl: in.API.TargetURL,
+		}
+
+		if in.API.Spec != nil {
+			outDeprecated.Api.ApiType = string(in.API.Spec.Type)
 		}
-	}
 
-	out.Labels = getLabelsOrNilIfEmpty(outLabels)
-	return out, nil
+		if in.API.DefaultAuth != nil && in.API.DefaultAuth.Credential != nil {
+			if outDeprecated.Api.Credentials == nil {
+				outDeprecated.Api.Credentials = &model.CredentialsWithCSRF{}
+			}
+			switch actual := in.API.DefaultAuth.Credential.(type) {
+			case graphql.BasicCredentialData:
+				outDeprecated.Api.Credentials.BasicWithCSRF = &model.BasicAuthWithCSRF{
+					BasicAuth: model.BasicAuth{
+						Username: actual.Username,
+						Password: actual.Password,
+					},
+				}
+			case graphql.OAuthCredentialData:
+				outDeprecated.Api.Credentials.OauthWithCSRF = &model.OauthWithCSRF{
+					Oauth: model.Oauth{
+						URL:          actual.URL,
+						ClientID:     actual.ClientID,
+						ClientSecret: actual.ClientSecret,
+					},
+				}
+			}
+		}
+
+		if in.API.DefaultAuth != nil && in.API.DefaultAuth.AdditionalHeaders != nil {
+			inHeaders := *in.API.DefaultAuth.AdditionalHeaders
+			outDeprecated.Api.Headers = &map[string][]string{}
+			if outDeprecated.Api.RequestParameters == nil {
+				outDeprecated.Api.RequestParameters = &model.RequestParameters{}
+			}
+			if outDeprecated.Api.RequestParameters.Headers == nil {
+				outDeprecated.Api.RequestParameters.Headers = &map[string][]string{}
+			}
+
+			for k, v := range inHeaders {
+				(*outDeprecated.Api.Headers)[k] = v
+				(*outDeprecated.Api.RequestParameters.Headers)[k] = v
+			}
+		}
+
+		if in.API.DefaultAuth != nil && in.API.DefaultAuth.AdditionalQueryParams != nil {
+			in := *in.API.DefaultAuth.AdditionalQueryParams
+			outDeprecated.Api.QueryParameters = &map[string][]string{}
+			if outDeprecated.Api.RequestParameters == nil {
+				outDeprecated.Api.RequestParameters = &model.RequestParameters{}
+			}
+
+			if outDeprecated.Api.RequestParameters.QueryParameters == nil {
+				outDeprecated.Api.RequestParameters.QueryParameters = &map[string][]string{}
+			}
+
+			for k, v := range in {
+				(*outDeprecated.Api.QueryParameters)[k] = v
```

Comment:
Just for better readability, can you define a separate variable for `*outDeprecated.Api.QueryParameters` value?